### PR TITLE
Roll up GetPrivilegesForColumns into GetColumnDefinitions

### PR DESCRIPTION
**Pull request by lavjain**
+ Simplify ACL logic to use Privileges string instead of ACL array
+ Remove columnPrivileges code that is no longer used
+ Update tests

### CODE REVIEW

File: backup/queries_table_defs.go [hunk 1]
```diff
@@ -271,26 +275,26 @@ func GetPrivilegesForColumns(connectionPool *dbconn.DBConn) map[uint32]map[strin
 		return metadataMap
```
Comment (lavjain) [inline backup/queries_table_defs.go]: `GetPrivilegesForColumns` is no longer necessary and can be removed.

File: backup/queries_acl.go [hunk 2]
```diff
@@ -285,10 +285,11 @@ ORDER BY n.nspname, a.defaclobjtype, r.rolname`
 }
 
 func GetQuotedRoleNames(connectionPool *dbconn.DBConn) map[string]string {
-	quotedRoleNames := make(map[string]string)
-
-	quotedRoleNamesQuery := `
-SELECT rolname AS rolename, quote_ident(rolname) AS quotedrolename FROM pg_authid`
+	if len(quotedRoleNames) > 0 {
```
Comment (lavjain) [inline backup/queries_acl.go]: This code will change once quotedroles PR is merged into master.

**Comment by jimmyyih** [comment]
Seems like there's a merge conflict with a PR you merged in yesterday.  Could you resolve it before we review?

**Comment by lavjain** [comment]
Merge conflicts have been resolved, Thanks!

File: backup/queries_table_defs.go [hunk 3]
```diff
@@ -162,12 +160,13 @@ type ColumnDefinition struct {
 	StorageType           string
 	DefaultVal            string
 	Comment               string
+	Privileges            sql.NullString
```
Comment (kyeap-pivotal) [inline backup/queries_table_defs.go]: This is a cool way to deal with NULL strings I didn't know of. In a lot of parts in our code opted to use coalesce.

File: integration/predata_table_defs_queries_test.go [hunk 4]
```diff
@@ -175,43 +168,13 @@ CREATE TABLE public.test_tsvector (
...
-
-			metadataMap := backup.GetPrivilegesForColumns(connectionPool)
-
-			oid := testutils.OidFromObjectName(connectionPool, "public", "granted_privileges", backup.TYPE_RELATION)
-			expectedACL := []backup.ACL{{Grantee: "testrole", Select: true}}
-			Expect(metadataMap).To(HaveLen(1))
-			Expect(metadataMap[oid]).To(HaveLen(1))
-			Expect(metadataMap[oid]["i"]).To(Equal(expectedACL))
-		})
-	})
```
Comment (kyeap-pivotal) [inline integration/predata_table_defs_queries_test.go]: We are removing the old integration tests verifying `GetPrivilegesForColumns` gets the correct privileges, but I don't see any new tests to verify the new code in `GetColumnDefinitions` is correct.

Comment (lavjain) [reply integration/predata_table_defs_queries_test.go]: Just added :-)

File: backup/predata_acl.go [hunk 5]
```diff
@@ -124,6 +125,21 @@ func ConstructMetadataMap(results []MetadataQueryStruct) MetadataMap {
...
+		privilegesStr = privileges.String
+	}
+	columnMetadata := make([]ACL, 0)
+	acl := ParseACL(privilegesStr)
+	if acl != nil {
+		columnMetadata = append(columnMetadata, *acl)
+	}
+	return columnMetadata
+}
+
```
Comment (lavjain) [inline backup/predata_acl.go]: Note: The logic for getColumnACL has been extracted from ConstructColumnPrivillegesMap function where it was being called in a loop for all columns.

File: backup/predata_relations_other_test.go [hunk 6]
```diff
@@ -479,74 +476,4 @@ GRANT ALL ON shamwow.shazam TO testrole;`,
...
-			privileges = []backup.ColumnPrivilegesQueryStruct{colEmpty}
-			metadataMap := backup.ConstructColumnPrivilegesMap(privileges)
-
-			expectedACLForEmptyKind := []backup.ACL{{Grantee: "GRANTEE"}}
-
-			Expect(metadataMap).To(HaveLen(1))
-			Expect(metadataMap[2]).To(HaveLen(1))
-			Expect(metadataMap[2]["m"]).To(Equal(expectedACLForEmptyKind))
-		})
-	})
```
Comment (lavjain) [inline backup/predata_relations_other_test.go]: `ConstructColumnPrivilegesMap` function is no longer used and has been removed

File: backup/queries_table_defs.go [hunk 7]
```diff
@@ -162,12 +160,13 @@ type ColumnDefinition struct {
...
+	Kind                  string
 	Options               string
 	FdwOptions            string
 	Collation             string
 	SecurityLabelProvider string
 	SecurityLabel         string
-	ACL                   []ACL
```
Comment (lavjain) [inline backup/queries_table_defs.go]: ACL array in `ColumnDefinition` structure has been replaced with Privileges and Kind fields

File: backup/queries_table_defs.go [hunk 8]
```diff
@@ -177,7 +176,7 @@ var storageTypeCodes = map[string]string{
 	"x": "EXTENDED",
 }
 
-func GetColumnDefinitions(connectionPool *dbconn.DBConn, columnMetadata map[uint32]map[string][]ACL) map[uint32][]ColumnDefinition {
+func GetColumnDefinitions(connectionPool *dbconn.DBConn) map[uint32][]ColumnDefinition {
```
Comment (lavjain) [inline backup/queries_table_defs.go]: The logic for getting column privileges has been moved into `GetColumnDefinition`

**Pull request merged**